### PR TITLE
Made a first pass at making the script cross platform

### PR DESCRIPTION
**Pull request by hilts-vaughan**
- Script has been tested on OSX, Arch Linux and Strawberry Perl on
  Windows

### CODE REVIEW

**Comment by hilts-vaughan** [comment]
Oh, I'd like to note that I removed the CRC32 dependency since it required GCC to build on Windows and GCC isn't very nice to setup on Windows -- and a lot of people simply won't have it installed. It seems like a superflous requirement when a simple native Perl implementation will do the trick.

**Comment by gnoling** [comment]
There are too many changes in this request, using strings rather then a list when invoking programs is an issue. It introduces both security and other potential issues. Is that required for it to work with windows?

I'm fine with invoking the programs off of the environment path, I'll just add a settings file later to customize that.

**Comment by hilts-vaughan** [comment]
I cleaned it up a bit since doing this to make it a bit more secure. I'll re-upload the changes; the main issue is with how Windows handles the paths. 

You may also be interested in this commit on my fork:

https://github.com/hilts-vaughan/UnlinkMKV-GUI/commit/2dd7252177b37722a78a689f174466a789c3f828

It resolves an issue where some MKVs like Coalgirl's releases break due to faulty MKV containers. I've removed the 0:00:00 check as it seemed redundant to me as well but I'm unsure why it was there as it was uncommented.

I'm new to Perl, so the code is fairly sloppy.

**Comment by gnoling** [comment]
The 0:00:00 check is also due to edge cases, if you don't need to remove it, don't.

I can see the need for multiple default chapters in certain scenarios, so that could actually be an unlinkmkv bug. When I get some time I'll check out their Lain release and see what's up.

**Comment by hilts-vaughan** [comment]
It's only on one episode, which makes me think it's a bug. I don't see how
the player could possibly have multiple default editions, default is what
plays first, right? In Lain, the two editions are actually dupes of each
other, hence thinking it's a bug. It also only happens on episode one.
Removing 0:00:00 fixes Lain Episode 13, which has an issue.

Can you think of an edge case where that check is actually needed? Or a
release? All mine worked.

**Comment by gnoling** [comment]
Unfortunately no, these fixes were added years ago and I never recorded what it was. Nearest I can find is a later adjustment to the fix to support a Turn-A Gundam release.

I believe it has to do with releases where the external segment is at the very beginning, when no split piece is in front of it and some sort of problem with detecting that properly.

**Comment by hilts-vaughan** [comment]
Weird; I'll try some more releases I don't have and see if any issues crop
up. I test mostly Coalgirl's releases.

**Comment by gnoling** [comment]
The Lain episode 13 problem led me to discover a different bug in UnlinkMKV, now fixed in 677e1b49d2e6b96bf2e992713746a7dcd927a6c8.

For episode 13 it's easy to correct the chapter issue that causes it in the first place, which is as simple as:

Open the file in mkvmerge GUI -> Chapter Editor -> Load -> Select the file -> Click "Opening" under "Edition" -> change 00:00:16 under start to 00:00:00 -> Chapter Editor -> Save to Matroska file -> Pick the original file. After that it'll work properly.

**Comment by hilts-vaughan** [comment]
The file is valid, though. Users shouldn't need to edit it to make it work.
:(

Thanks for the tip, though.

**Comment by gnoling** [comment]
As near as I can tell, the correct start time for that chapter is
00:00:00. The _correct_ fix is patching the file -in this case-.

That said, it's not incorrect on a technical level either. In that
specific case, the chapters are wrong and are easily fixed, but if they
weren't, UnlinkMKV wouldn't support that file. Currently UnlinkMKV can
only process chapters correctly on a single level, without additional
splits to any splits / additional chapter or segment processing.
Basically external segment support is limited to "no additional
chapters, and only the entire segments".

-Garret

> The file is valid, though. Users shouldn't need to edit it to make it
> work.
> :(
> 
> Thanks for the tip, though.
